src/cmd/serve.rs

```diff
@@ -116,38 +119,70 @@ async fn serve(
     reload_tx: broadcast::Sender<Message>,
     file_404: &str,
 ) {
-    // A warp Filter which captures `reload_tx` and provides an `rx` copy to
-    // receive reload messages.
-    let sender = warp::any().map(move || reload_tx.subscribe());
-
-    // A warp Filter to handle the livereload endpoint. This upgrades to a
-    // websocket, and then waits for any filesystem change notifications, and
-    // relays them over the websocket.
-    let livereload = warp::path(LIVE_RELOAD_ENDPOINT)
-        .and(warp::ws())
-        .and(sender)
-        .map(|ws: warp::ws::Ws, mut rx: broadcast::Receiver<Message>| {
-            ws.on_upgrade(move |ws| async move {
-                let (mut user_ws_tx, _user_ws_rx) = ws.split();
-                trace!("websocket got connection");
-                if let Ok(m) = rx.recv().await {
-                    trace!("notify of reload");
-                    let _ = user_ws_tx.send(m).await;
+    let reload_tx_clone = reload_tx.clone();
+
+    // WebSocket handler for live reload
+    let websocket_handler = move |ws: WebSocketUpgrade| async move {
+        let reload_tx = reload_tx_clone.clone();
+        ws.on_upgrade(move |socket| websocket_connection(socket, reload_tx))
+    };
+
+    // Serve static files with custom 404 fallback
+    let file_404_path = build_dir.join(file_404);
+    let serve_dir = ServeDir::new(&build_dir).not_found_service(tower::service_fn(
+        move |_: axum::http::Request<axum::body::Body>| {
```

Comment:
Thanks!

There is a similar PR to switch to axum in https://github.com/rust-lang/mdBook/pull/2359. That takes a slightly different approach for handling the `not_found_service` which seems a bit simpler:

```rust
.not_found_service(ServeFile::new(build_dir.join(file_404)))
```

Can you say more about why the difference here and why this needs to do mime guessing?